overlord/snapstate/snapstate.go

```diff
@@ -2000,6 +2003,36 @@ func autoRefreshPhase1(ctx context.Context, st *state.State) ([]string, []*state
 		}
 	}
 
+	// if specific gating snap was given, drop other affected snaps unless
+	// they are affected by same updates as forGatingSnap.
+	if forGatingSnap != "" {
+		if _, ok := affectedSnaps[forGatingSnap]; !ok {
+			// this can happen if after refreshing candidates the snap is no
+			// longer affected by new candidates.
+			return nil, nil, nil
+		}
+		// filter out unrelated affectedSnaps
+		for gatingSnap, affectedInfo := range affectedSnaps {
+			if gatingSnap == forGatingSnap {
+				continue
+			}
+			var found bool
+			for affectingSnap := range affectedInfo.AffectingSnaps {
+				// does this affectingSnap also affect forGatingSnap?
+				// if so, we will keep it around, meaning we will run its
+				// gate-auto-refresh hook because it may potentially hold updates
+				// even if forGatingSnap's hook says "proceed".
+				if _, ok := affectedSnaps[forGatingSnap].AffectingSnaps[affectingSnap]; ok {
```

Comment:
we don't need the ok bit because this is a map of bools?  also  affectedSnaps[forGatingSnap].AffectingSnaps is fixed for both the loops so we could read it earlier outside? they are the affecting snaps for the forGatingSnap which basically is the snap we will try to update here?

Reply:
Indeed, simplified.